### PR TITLE
ARC-819: remove rate-limiting logic on blanket 403 and instead log an…

### PR DESCRIPTION
**Pull request by thombergs**

### CODE REVIEW

File: src/config/enhance-octokit.ts [hunk 1]
```diff
@@ -29,11 +29,11 @@ const instrumentRequests = (octokit: GitHubAPI) => {
 			throw new RateLimitingError(rateLimitReset);
 		}
 
-		if (error.status === 403) {
-			// delaying for an hour
-			logger.warn({ err: error }, "GitHub returned 403");
-			throw new RateLimitingError(new Date().getTime() / 1000 + 60 * 60);
```
Comment (thombergs) [inline src/config/enhance-octokit.ts:35]: previously, we would throw a `RateLimitingError` on every 403 from GitHub, which is wrong

File: src/config/enhance-octokit.ts [hunk 2]
```diff
@@ -29,11 +29,11 @@ const instrumentRequests = (octokit: GitHubAPI) => {
...
+		if (error.status === 403 && error.stack?.includes("has an IP allow list enabled")) {
+			logger.warn({ err: error }, "blocked by GitHub allowlist!");
```
Comment (mboudreau) [inline src/config/enhance-octokit.ts:33]: I'd probably make that an error, not a warn

File: src/config/enhance-octokit.ts [hunk 3]
```diff
@@ -29,11 +29,11 @@ const instrumentRequests = (octokit: GitHubAPI) => {
...
+			statsd.increment(metricError.blockedByGitHubAllowlist);
 		}
```
Comment (thombergs) [inline src/config/enhance-octokit.ts:35]: Now, we're checking the stack trace for a special error message from GitHub and emit a metric to make IP allowlist errors more prominent (I'll later add a low-prio alert on that metric)

File: src/config/enhance-octokit.ts (hunk 2 above)
Comment (thombergs) [reply src/config/enhance-octokit.ts]: done

File: src/config/enhance-octokit.ts (hunk 3 above)
Comment (thombergs) [reply src/config/enhance-octokit.ts:35]: Tested the metric and new log event in staging.

**Review by rachellerathbone — APPROVED** [review]

**Review by atrigueiro — APPROVED** [review]

**Pull request merged**